### ISSUE TITLE
Using expression for base-url throws when enabling CircuitBreaker

### ISSUE THREAD

**Issue by jrehwaldt** [zalando/riptide#843]
When enabling the circuit-breaker integration *and* using an expression for the base-url the application fails to start with an exception:

```
java.lang.IllegalArgumentException: Illegal character in scheme name at index 0: ${GATEWAY_BASE_URL}
```

## Description
The following setup causes riptide to fail during initialization when setting `GATEWAY_BASE_URL` as environment variable. Disabling circuit-breaker will allow riptide to initialize properly.

```
riptide:
  clients:
    test:
      circuit-breaker:
        enabled: true
      base-url: ${GATEWAY_BASE_URL}
```

The problem seems to be that during execution phase of `DefaultRiptideRegistrar` the properties are either not yet expanded by Spring or the way they are accessed is incorrect. This causes [`DefaultRiptideRegistrar#getHost`](
https://github.com/zalando/riptide/blob/619c71b8ad3536c0bc3842d0ae3fd68e2e7ee687/riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideRegistrar.java#L554-L561) to receive the un-expanded property expression `${GATEWAY_BASE_URL}`.

```
java.lang.IllegalArgumentException: Illegal character in scheme name at index 0: ${GATEWAY_BASE_URL}
	at java.net.URI.create(URI.java:852)
	at java.util.Optional.map(Optional.java:215)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.getHost(DefaultRiptideRegistrar.java:560)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.clientName(DefaultRiptideRegistrar.java:555)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.lambda$registerCircuitBreakerListener$36(DefaultRiptideRegistrar.java:542)
	at org.zalando.riptide.autoconfigure.Registry.lambda$registerIfAbsent$1(Registry.java:37)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:36)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:32)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.registerCircuitBreakerListener(DefaultRiptideRegistrar.java:537)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.lambda$registerCircuitBreaker$35(DefaultRiptideRegistrar.java:533)
	at org.zalando.riptide.autoconfigure.Registry.lambda$registerIfAbsent$1(Registry.java:37)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:36)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:32)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.registerCircuitBreaker(DefaultRiptideRegistrar.java:529)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.lambda$registerCircuitBreakerFailsafePlugin$25(DefaultRiptideRegistrar.java:422)
	at org.zalando.riptide.autoconfigure.Registry.lambda$registerIfAbsent$1(Registry.java:37)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:36)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.registerCircuitBreakerFailsafePlugin(DefaultRiptideRegistrar.java:417)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.registerPlugins(DefaultRiptideRegistrar.java:258)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.lambda$registerHttp$1(DefaultRiptideRegistrar.java:124)
	at org.zalando.riptide.autoconfigure.Registry.lambda$registerIfAbsent$1(Registry.java:37)
	at java.util.Optional.orElseGet(Optional.java:267)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:36)
	at org.zalando.riptide.autoconfigure.Registry.registerIfAbsent(Registry.java:32)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.registerHttp(DefaultRiptideRegistrar.java:114)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.lambda$register$0(DefaultRiptideRegistrar.java:107)
	at com.google.common.collect.SingletonImmutableBiMap.forEach(SingletonImmutableBiMap.java:65)
	at org.zalando.riptide.autoconfigure.DefaultRiptideRegistrar.register(DefaultRiptideRegistrar.java:106)
	at org.zalando.riptide.autoconfigure.RiptidePostProcessor.postProcessBeanDefinitionRegistry(RiptidePostProcessor.java:38)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:275)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:125)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:706)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:532)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
	at xxxxxxx
Caused by: java.net.URISyntaxException: Illegal character in scheme name at index 0: ${GATEWAY_BASE_URL}
	at java.net.URI$Parser.fail(URI.java:2848)
	at java.net.URI$Parser.checkChars(URI.java:3021)
	at java.net.URI$Parser.checkChar(URI.java:3031)
	at java.net.URI$Parser.parse(URI.java:3047)
	at java.net.URI.<init>(URI.java:588)
	at java.net.URI.create(URI.java:850)
	... 47 common frames omitted
```

## Expected Behavior
Riptide initialized properly using variable expressions and circuit-breaker together.

## Actual Behavior
It fails with `java.net.URISyntaxException: Illegal character in scheme name [...]`.

## Possible Fix
* Try-catch and fallback to `id` in error case (currently only in case `base-url` is not set at all)
* Ensure properties are correctly initialized when accessing
* Allow to manually override the id via a property `circuit-breaker.client-id: my-customer-id` (seems pointless, since the riptide client has already an id)

## Your Environment
Spring Boot 2.2.1.RELEASE
Riptide 3.0.0.RC8 (judging from the code also earlier 3.0.0.RC versions are affected)

**Comment by whiskeysierra** [comment]
Thanks for the issue. I opened #844 with a fix.

**Comment by whiskeysierra** [comment]
Fixed in [3.0.0-RC.9](https://github.com/zalando/riptide/releases/tag/3.0.0-RC.9).
